core/src/lib/page-object/page-object-loader.ts

```diff
@@ -58,7 +58,7 @@ function setResultAttributes(
     result.hasFillForm = params.hasFillForm;
     result.generatedPageObjectPath = params.generatedPageObjectPath;
     result.generatedExtendingPageObjectPath = params.generatedExtendingPageObjectPath;
-    result.historyUid = params.pageObjectBuilder.historyUidCounter;
+    result.historyUid = params.useHistoryUid !== undefined ? params.useHistoryUid : params.pageObjectBuilder.historyUidCounter;
```

Comment:
Why the concrete check !== undefined?

params.useHistoryUid ? ... : ... isn't enough?

Reply:
No because if the Uid would be 0 the truthiness check would be also false.

Reply:
I suspected that, so everythong fine!